src/core/GAT.jl

```diff
@@ -188,11 +189,16 @@ end
 
 function parse_theory_head(expr::Expr)::TheoryHead
   parse = parse_theory_binding
+  parse_jl = parse_theory_binding_jlstyle
+  parse_either = parse_theory_binding_either
   @match expr begin
     (Expr(:call, :(=>), Expr(:tuple, bases), main)
       => TheoryHead(parse(main), map(parse, bases)))
+    (Expr(:(<:), main, Expr(:tuple,bases))
+      => TheoryHead(parse_jl(main), map(parse_jl, bases)))
     Expr(:call, :(=>), base, main) => TheoryHead(parse(main), [parse(base)])
-    _ => TheoryHead(parse(expr))
+    Expr(:(<:), main, base) => TheoryHead(parse_jl(main), [parse_jl(base)])
+    _ => TheoryHead(parse_either(expr))
```

Comment:
Maybe you can avoid having to define `parse_either` by replacing this last case with two cases:

```julia
Expr(:call, _...) => TheoryHead(parse(expr))
_ => TheoryHead(parse_jl(expr))
```

Reply:
Good call.

Reply:
Oh, you mean in `@instance`, not `@syntax`.